packages/core/src/shared/errors.ts

```diff
@@ -584,3 +588,17 @@ export function isNetworkError(err?: unknown): err is Error & { code: string } {
         'EADDRNOTAVAIL', // port not available/allowed?
     ].includes(err.code)
 }
+
+/**
+ * This error occurs on a network call if the user has set up a proxy in the
+ * VS Code settings but the proxy is not reachable.
+ *
+ * Setting ID: http.proxy
+ */
+function isVSCodeProxyError(err?: unknown): boolean {
+    if (!(err instanceof Error)) {
+        return false
+    }
+
+    return err.name === 'Error' && err.message.startsWith('Failed to establish a socket connection to proxies')
```

Comment:
what other fields are set on the object? is there any other (non-standard) field we can use as a more reliable hint?

Reply:
This is the best we can do with the error that they give us. Nothing else useful I could see in it